crates/ruff_linter/src/rules/airflow/rules/dag_schedule_argument.rs

```diff
@@ -0,0 +1,113 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::Expr;
+use ruff_python_ast::{self as ast, Keyword};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks that the `DAG()` class or a `@dag()` decorator has an explicit
+/// `schedule` parameter.
+///
+/// ## Why is this bad?
+/// The default `schedule` value on Airflow 2 is `timedelta(days=1)`, which is
+/// almost never what a user is looking for. Airflow 3 changes this the default
+/// to *None*, and would break existing DAGs using the implicit default.
+///
+/// Airflow 2 also provides alternative arguments `schedule_interval` and
+/// `timetable` to specify the DAG schedule. They existed for backward
+/// compatibility, and have been removed from Airflow 3.
```

Comment:
I'm a bit torn about whether this is something that belongs to this rule or it is a better fit for the `Airflow3Deprecation` rule. I can see how it fits here because it is `DAG` related but it feels a bit out of scope reading through the rule's description (It isn't mentioned at all in the *What it does* section)

Reply:
I've a slight preference to move it into the other deprecation rule or make it its own rule. 

For my understanding, are `schedule_interval`, and `timetable` now deprecated or have they been deprecated and were removed as part of airflow 3? Depending on the answer, I think a separate rule makes more sense

Reply:
I suggest moving them to a separate rule. Maybe one that catches all Airflow 3 removals?

Reply:
Got it, I’ll move them to #14582 (AIR302) then.

Reply:
Alright done.